sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -762,6 +877,53 @@ dynamic parameter for the buffer can be registered with all the nodes which
 use the buffer as a parameter. Then a single `dynamic_parameter::update()` call
 will maintain the graphs data dependencies.
 
+====== Execution Range Updates
+
+Another configuration that can be updated is the execution range of the
+kernel, this can be set through `node::update_nd_range()` or
+`node::update_range()` but does not require any prior registration.
+
+An alternative way to update the execution range of a node is to do so while
+updating command groups as described in the next section. Using this mechanism
+lifts the restriction from `node::update_nd_range()` / `node::update_range()`
+of only being to update the execution range in the same dimension. As the
+update being tied to a change in command-group means that the updated kernel
+code may be defined as operating in a different dimension.
+
+====== Command Group Updates
+
+The command-groups of a kernel node can be updated using dynamic command-groups.
+Dynamic command-groups allow replacing the command-group function of a kernel
+node with a different one. This effectively allows updating the kernel function
+and/or the kernel execution range.
+
+Command-group updates are performed by creating an instance of the
+`dynamic_command_group` class. A dynamic command-group is created with a modifiable
+state graph and a list of possible command-group functions. Command-group functions
+within a dynamic command-group can then be set to active by using the member function
+`dynamic_command_group::set_active_cgf()`.
+
+Dynamic command-groups are compatible with dynamic parameters. This means that
+dynamic parameters can be used in command-group functions that are part of
+dynamic command-groups. Updates to such dynamic parameters will be reflected
+in the command-group functions once they are activated.
+
+Note that the execution range is tied to the command-group, therefore updating
+the range of a node which uses a dynamic command-group shared by another node
+will also update the execution range of the nodes sharing the dynamic
+command-group. Activating a command-group with `set_active_cgf` to a
+command-group that previously had its execution range updated with
+`node::update_range()` or `node::update_nd_range()` will not reset the execution
+range to the original, but instead use the most recently updated value.
```

Comment:
Could we mention explicitly that updating the range of a node using a dynamic command-group will only update the range of the currently active CGF? It is sort of implied by talking about the fact it won't be reset, but I think it would be good for clarity to specify it explicitly first.

Reply:
Good shout, mentioned this explicitly